src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -955,6 +955,18 @@ void primitive_inst::realloc_if_needed(bool prev_execution_skipped) {
                                           is_output_buffer(this, true),
                                           output_memory_ptr(i).get(),
                                           true);
+            // memory_pool does not reset padded memory to be reused.
+            if (static_cast<bool>(_outputs[i]->get_layout().data_padding)
+                && need_reset_output_memory() && !can_be_optimized()) {
+                auto mem_ptr = _outputs[i];
+                auto num_users = memory_pool::count_users_in_padded_pool(_network.get_memory_pool(), mem_ptr);
+                // If the memory is already being used in more than 1 layer, it need to reset.
+                if (num_users > 1) {
+                    GPU_DEBUG_TRACE_DETAIL << id() << " : Need reset padded output memory considering user" << std::endl;
+                    add_dep_event(_outputs[i]->fill(_network.get_stream()));
+                }
```

Comment:
Does this mean that the first allocation call will be performed without memory reset, and all the subsequent iterations will reuse this "dirty" buffer if the users count is not changed?

Reply:
Yes. The first allocation does not require a memory reset, and since the same layer is likely to be used over and over again if the users count is not increased, the buffer will not get dirty. So I think it is okay to reuse it. Am I missing something?

Reply:
Could you please clarify why the first allocation doesn't require memory reset? I assume the implementation relies on it, as we have need_reset_output_memory=true

Reply:
The first allocation calls `ocl_engine::allocate_memory()`. The memory is reset in this function. https://github.com/openvinotoolkit/openvino/blob/3ceeeb52d643e564f50d9a6f400561104528e828/src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp#L191